x-pack/plugins/ml/public/application/data_frame_analytics/common/fields.ts

```diff
@@ -267,38 +274,37 @@ export const getDefaultFieldsFromJobCaps = (
   const featureImportanceFields = [];
 
   if ((numTopFeatureImportanceValues ?? 0) > 0) {
-    featureImportanceFields.push(
-      ...fields.map(d => ({
-        id: `${resultsField}.feature_importance.${d.id}`,
-        name: `${resultsField}.feature_importance.${d.name}`,
-        type: KBN_FIELD_TYPES.NUMBER,
-      }))
-    );
+    featureImportanceFields.push({
+      id: `${resultsField}.feature_importance`,
+      name: `${resultsField}.feature_importance`,
+      type: KBN_FIELD_TYPES.NUMBER,
+    });
```

Comment:
I'm not sure I understand this change, can you explain why having the individual fields is no longer necessary?

Reply:
The `feature_importance` field is an array of objects so the `<results_field>.feature_importance.<id>` field name wasn't actually mapping to anything - that column was just empty.